mlflow/models/evaluation/base.py

```diff
@@ -793,52 +793,90 @@ def evaluate(
     :param custom_metrics: (Optional) A list of custom metric functions. A custom metric
                            function is required to take in two parameters:
 
-                           - Union[pandas.Dataframe, pyspark.sql.DataFrame]: The first being a
+                           - ``Union[pandas.Dataframe, pyspark.sql.DataFrame]``: The first being a
                              Pandas or Spark DataFrame containing ``prediction`` and ``target``
                              column. The ``prediction`` column contains the predictions made by
                              the model. The ``target`` column contains the corresponding labels
                              to the predictions made on that row.
-                           - Dict: The second is a dictionary containing the metrics calculated
+                           - ``Dict``: The second is a dictionary containing the metrics calculated
                              by the default evaluator. The keys are the names of the metrics
                              and the values are the scalar values of the metrics. Refer to the
                              DefaultEvaluator behavior section for what metrics will be returned
                              based on the type of model (i.e. classifier or regressor).
 
                            A custom metric function can return in the following format:
 
-                           - Dict[AnyStr, Union[int, float, np.number]: a singular dictionary of
+                           - ``Dict[AnyStr, Union[int, float, np.number]``: a singular dictionary of
                              custom metrics, where the keys are the names of the metrics, and the
                              values are the scalar values of the metrics.
+                           - ``Tuple[Dict[AnyStr, Union[int,float,np.number]], Dict[AnyStr,Any]]``:
+                             a tuple of a dict containing the custom metrics, and a dict of
+                             artifacts, where the keys are the names of the artifacts, and the
+                             values are objects representing the artifacts.
+
+                           Object types that artifacts can be represented as:
+
+                           - A string uri representing the file path to the artifact. MLflow will
+                             infer the type of the artifact based on the file extension.
+                           - A string representation of a JSON object. This will be saved as a
+                             .json artifact.
+                           - ``EvaluationArtifact`` type objects. i.e. ``CsvEvaluationArtifact``,
+                             ``ImageEvaluationArtifact``.
```

Comment:
Can we explain that, in these cases, the user is expected to log the associated artifact content to MLflow manually from within the `custom_metric` function?